### DISCUSSION TITLE
Progress Bar Updates

### DISCUSSION THREAD

**Discussion by b202i** [Studio-42/elFinder#3311] [Ideas]
Is there a way for a Connector to communicate progress back to elFinder in order to update the Progress Bar for a long running task?  In this case it is for Extract files from archive (pic attached).  This would be useful for just about any task that involves multiple files.

![progress bar long running connector task](https://cloud.githubusercontent.com/assets/16022959/14921527/ab0c92ba-0e00-11e6-8ef8-063564f57d2c.png)

**Comment by nao-pon** [comment]
@ma2tkraus I also think it is necessary, but not yet implemented. Please to PR if there is a good idea.

**Comment by troex** [comment]
Also it will mean we have to parse output of different archivers in order to get current progress and I'm not sure it's easy and right task. Also additional problem that we will have to write current progress state into some temporary file, so `status pull` request could read from it.

There are no yet plans to implement it

**Comment by CodeLyokoXtEAM** [comment]
@nao-pon 
This site is is update automatically uploaded progress. So I search about it & they use "websocket-server-and-client-communication" to communicate sever & client in using wss request without ending connection.

![image](https://user-images.githubusercontent.com/18566890/40883932-3a8a472c-6726-11e8-8d07-ce71d109575f.png)

![image](https://user-images.githubusercontent.com/18566890/40883938-507b8df2-6726-11e8-8632-8d78b5e2112c.png)

http://cuelogic.com/blog/php-and-html5-websocket-server-and-client-communication/
https://tools.ietf.org/html/rfc6455
